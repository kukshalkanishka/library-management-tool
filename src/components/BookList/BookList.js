import React from "react";
import Book from "../Book/Book";

class BookList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { books: props.books };
    this.toggleTodoState = this.toggleTodoState.bind(this);
    this.deleteTodo = this.deleteTodo.bind(this);
  }

  async toggleTodoState(body) {
    await fetch("/toggleTodoState", {
      method: "PATCH",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(body)
    })
      .then(res => res.json())
      .then(res => {
        this.setState({ books: res });
      });
  }

  async deleteTodo(body) {
    await fetch("/todo", {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(body)
    })
      .then(res => res.json())
      .then(res => {
        this.setState({ books: res });
      });
  }

  componentWillReceiveProps(newProps) {
    this.setState({
      books: newProps.books
    });
  }

  render() {
    const {books} = this.state;
    return (
      <div>
        {Object.keys(books).reverse().map(bookName => {
          return (
            <Book
              key={books[bookName].id}
              details={books[bookName]}
              name={bookName}
              toggleTodoState={this.toggleTodoState}
              deleteTodo={this.deleteTodo}
            />
          );
        })}
      </div>
    );
  }
}

export default BookList;
