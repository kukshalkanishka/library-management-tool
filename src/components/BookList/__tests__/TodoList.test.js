import React from "react";
import { shallow, configure } from "enzyme";
import BookList from "../BookList";

import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("BookList", () => {
  it("should render properly", () => {
    const todos = [{ todo: "dummyTodo", time: "123" }];
    const toggleTodoState = jest.fn();
    const deleteTodo = jest.fn();
    const wrapper = shallow(
      <BookList
        todos={todos}
        toggleTodoState={toggleTodoState}
        deleteTodo={deleteTodo}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
