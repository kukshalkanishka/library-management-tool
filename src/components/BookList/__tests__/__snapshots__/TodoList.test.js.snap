// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render properly 1`] = `
"<div>
  <Todo todo={{...}} toggleTodoState={[Function: bound toggleTodoState]} deleteTodo={[Function: bound deleteTodo]} />
</div>"
`;
